src/webui/api/rsscontroller.cpp

```diff
@@ -91,7 +94,28 @@ void RSSController::itemsAction()
     setResult(jsonVal.toObject());
 }
 
-void RSSController::refreshItemAction()
+void RSSController::markAsReadAction() const
+{
+    requireParams({"itemPath"});
+
+    const QString itemPath {params()["itemPath"]};
+    const QString articleId {params()["articleId"]};
+
+    RSS::Item *item = RSS::Session::instance()->itemByPath(itemPath);
+
+    if (!item) return;
+
+    if(!articleId.isEmpty()) {
+        RSS::Feed *feed = qobject_cast<RSS::Feed *>(item);
+        if (feed)
+            feed->articleByGUID(articleId)->markAsRead();
```

Comment:
Don't forget that any element may already be missing at the moment, so you should check it.